### ISSUE TITLE
enforce replaceAll over replace with regex global

### ISSUE THREAD

**Issue by belchior** [eslint/eslint#12557]
**Enforce replaceAll over replace with regex global**

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const queryString = 'q=query+string+parameters';
// less readable and performative
queryString.replace(/\+/g, ' ');

// readable and performative
queryString.replaceAll('+', ' ');
```

```js
const text = 'I like strawberry';
// less readable and performative
text.replace(/strawberry/g, '🍓')

// readable and performative
text.replaceAll('strawberry', '🍓');
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This rules will improve readability and performance of the code base

More info:
https://github.com/tc39/proposal-string-replaceall
https://v8.dev/features/string-replaceall

**Comment by aladdin-add** [comment]
Based on [our policy](https://github.com/eslint/eslint#what-about-experimental-features), eslint don't support stage-3 features officially, we can reconsider it when it reaches stage-4 someday.
